numpy/core/src/umath/logical_gufuncs.c.src

```diff
@@ -256,79 +229,376 @@ static char types[] = {NPY_BYTE, NPY_BYTE, NPY_BOOL,
+    "out : ndarray or bool\n"
+    "   Output array of bools, or a single bool if x1 and x2 are 1D.\n"
+    "\n"
+    "See Also\n"
+    "-------\n"
+    "greater, any, all_greater, any_less_equal\n"
+    "\n"
+    "Examples\n"
+    "-------\n"
+    ">>> np.any_greater(np.arange(3), np.arange(3))\n"
+    "False\n"
+    ">>> np.any_greater([[1, 2], [0, 0], [2, 3]], [1, 3])\n"
+    "array([ False, False,  True], dtype=bool)";
+
+static char const * const any_greater_equal_doc =
+    "Return True if x1 >= x2 for any elements along the last axis, False\n"
+    "otherwise.  Similar to (x1>=x2).any(axis=-1), except the last dimension\n"
+    "of x1 and x2 must be equal and greater than 1.\n"
+    "\n"
+    "Parameters\n"
+    "----------\n"
+    "x1, x2 : array_like\n"
+    "    Input arrays of the same shape.\n"
+    "\n"
+    "Returns\n"
+    "-------\n"
+    "out : ndarray or bool\n"
+    "   Output array of bools, or a single bool if x1 and x2 are 1D.\n"
+    "\n"
+    "See Also\n"
+    "-------\n"
+    "greater_equal, any, all_greater_equal, any_less\n"
+    "\n"
+    "Examples\n"
+    "-------\n"
+    ">>> np.any_greater_equal(np.arange(3), np.arange(3))\n"
+    "True\n"
+    ">>> np.any_greater_equal([[1, 2], [0, 0], [2, 3]], [1, 3])\n"
+    "array([ True, False,  True], dtype=bool)";
+
 
 /* -------------------------------------------------------------------------- */
 /* function to create and register all gufuncs*/
+
 void InitLogicalGufuncs(PyObject *dictionary, 
                         PyUFunc_FromFuncAndDataAndSignature_t createPyUFunc) {
     PyObject *f;
-    int i;
-    const int gufunc_count = sizeof(gufunc_descriptors)/
-        sizeof(gufunc_descriptors[0]);
-
-    for (i=0; i < gufunc_count; i++) {
-        GUFUNC_DESCRIPTOR_t* d = &gufunc_descriptors[i];
-
-        f = createPyUFunc(d->funcs,
-                          array_of_nulls,
-                          types,
-                          sizeof(types),  // number of types
-                          2,              // number of inputs
-                          1,              // number of outputs
-                          PyUFunc_None,
-                          d->name,
-                          "temp docstring",
-                          0,              // unused
-                          "(i),(i)->()");
-
-        PyDict_SetItemString(dictionary, d->name, f);
-        Py_DECREF(f);
-    }
+
+/**begin repeat
+* #NAME = all_equal,all_not_equal,all_less,all_less_equal,all_greater,all_greater_equal,
+*         any_equal,any_not_equal,any_less,any_less_equal,any_greater,any_greater_equal#
+* #APPROACH = dispatch,base*11#
+*/
+
+    static PyUFuncGenericFunction @NAME@_funcs[] = { 
+
+/**begin repeat1
+* #TYPE = byte,ubyte,short,ushort,int,uint,long,ulong,longlong,ulonglong,float,
+*         double,longdouble,cfloat,cdouble,clongdouble#
+*/
+
+        @APPROACH@_@TYPE@_@NAME@,
+
+/**end repeat1**/
+
+    }; // close array of funcs
+
+
+
+    f = createPyUFunc(@NAME@_funcs,
```

Comment:
I'm afraid this is  ISO C90, so this either needs to be in a separate `repeat` to `static PyUFuncGenericFunction @NAME@_funcs ...`, or the entire body of the `repeat` needs to be inside `{}`